applications/GeoMechanicsApplication/tests/cpp_tests/test_geomechanics_erosion_piping.cpp

```diff
@@ -0,0 +1,211 @@
+{
+public:
+    using BaseType = ImplicitSolvingStrategy<TSparseSpace, TDenseSpace, TLinearSolver>;
+    using TConvergenceCriteriaType = ConvergenceCriteria<TSparseSpace, TDenseSpace>;
+    using TBuilderAndSolverType    = typename BaseType::TBuilderAndSolverType;
+    using TSchemeType              = typename BaseType::TSchemeType;
+
+    MockGeoMechanicsNewtonRaphsonErosionProcessStrategy(ModelPart&                    model_part,
+                                                        typename TSchemeType::Pointer pScheme,
+                                                        typename TLinearSolver::Pointer pNewLinearSolver,
+                                                        typename TConvergenceCriteriaType::Pointer pNewConvergenceCriteria,
+                                                        typename TBuilderAndSolverType::Pointer pNewBuilderAndSolver,
+                                                        Parameters& rParameters)
+        : GeoMechanicsNewtonRaphsonErosionProcessStrategy<TSparseSpace, TDenseSpace, TLinearSolver>(
+              model_part, pScheme, pNewLinearSolver, pNewConvergenceCriteria, pNewBuilderAndSolver, rParameters),
+          mModelPart(model_part) {};
+
+private:
+    bool Recalculate() override
+    {
+        // The function provides a pressure change over iterations that leads to a non-linear growth of an equilibrium pipe height.
+        // As a result the height curve crosses twice a pipe height growth curve in the search algorithm implemented
+        // in  GeoMechanicsNewtonRaphsonErosionProcessStrategy. The search stops at the second point.
+        for (auto& node : mModelPart.Nodes()) {
+            node.FastGetSolutionStepValue(WATER_PRESSURE) =
+                pow(node.FastGetSolutionStepValue(WATER_PRESSURE), 0.9725);
+        }
+        return true;
+    }
+
+    void BaseClassFinalizeSolutionStep() override {
+        // to avoid calling the BaseClassFinalizeSolutionStep in tested class.
+    };
+    ModelPart& mModelPart;
+};
+
+Node::Pointer CreateNodeWithFastSolutionStepValues(
+    ModelPart& rModelPart, int Id, double CoordinateX, double CoordinateY, double WaterPressure)
+{
+    constexpr double coordinate_z = 0.0;
+    auto             p_node = rModelPart.CreateNewNode(Id, CoordinateX, CoordinateY, coordinate_z);
+
+    p_node->SetLock();
+
+    p_node->FastGetSolutionStepValue(WATER_PRESSURE) = WaterPressure;
+    const array_1d<double, 3> GravitationalAcceleration{0, 9.81, 0};
+    p_node->FastGetSolutionStepValue(VOLUME_ACCELERATION) = GravitationalAcceleration;
+
+    p_node->UnSetLock();
+
+    return p_node;
+}
+
+Geometry<Node>::Pointer CreateQuadrilateral2D4N(
+    ModelPart& rModelPart, std::vector<int> NodeIds, double Xmin, double Xmax, double WaterPressureLeft, double WaterPressureRight)
+{
+    constexpr double y_min = -0.1;
+    constexpr double y_max = 0.1;
+
+    return Kratos::make_shared<Quadrilateral2D4<Node>>(
```

Comment:
I believe the namespace qualification is redundant here, since the mock class is in namespace `Kratos`:
```suggestion
    return make_shared<Quadrilateral2D4<Node>>(
```

Reply:
removed

Reply:
Apparently, it needs to stay there. Not sure why, but it seems I was wrong. Please ignore this suggestion.